src/sanity/sanity-checks.ts

```diff
@@ -0,0 +1,271 @@
+import microsoftgraph from '../modules/microsoftgraph/microsoftgraph.module'
+
+class Skipped extends CustomError {
+  public constructor (
+    message?: string,
+  ) {
+    super(message)
+  }
+}
+
+const check = (name: string, fn: () => Promise<void>, abortOnFail = false) => {
+  return {
+    name: name,
+    fn: fn,
+    abortOnFail: abortOnFail
+  }
+}
+
+const checkSuite = (name, checks) => {
+  return {
+    name: name,
+    checks: checks
+  }
+}
+
+const configSuite = () => {
+  const checks = [
+    check('Server options are set', async () => {
+      // check that either http or https (or both) are set
+      const [httpPort, httpsPort, sslKey, sslCert] = await Promise.all([appConfig.httpPort, appConfig.httpsPort, appConfig.sslKey, appConfig.sslCert])
+      if (!httpPort && !httpsPort) {
+        throw Error('Http port nor https port is set.')
+      }
+      if (httpsPort && (!sslKey)) {
+        throw Error('No SSL key set. Either set it or disable https.')
+      }
+      if (httpsPort && (!sslCert)) {
+        throw Error('No SSL cert set. Either set it or disable https.')
+      }
+    }, true),
+    check('Can read secrets from Azure Vault', async () => {
+      const vaultName = process.env.AZURE_KEY_VAULT_NAME
+      if (!Boolean(vaultName)) {
+        throw new Skipped('Azure Key Vault not configured.')
+      }
+      const credential = new DefaultAzureCredential()
+      const vaultUrl = `https://${vaultName}.vault.azure.net`
+      const client = new SecretClient(vaultUrl, credential)
+      try {
+        await client.getSecret('dummy')
+      } catch (err) {
+        if (err.name && err.name.includes('AuthenticationError')) {
+          throw Error(`Authentication error. See details below.\n${err.message}`)
+        } else if (err.name === 'RestError') {
+          // no error except 404 is allowed
+          if (!err.statusCode || err.statusCode != 404) {
+            throw Error(`Unable to read a secret. See below for details.\n${err.message}`)
+          }
+        } else {
+          throw Error(`Unable to read a secret. See below for details.\n${err.message}`)
+        }
+      }
+    }, true),
+    check('Can read secrets from Google Secret Manager', async () => {
+      const gsmProject = process.env.GCP_SECRET_MANAGER_PROJECT_ID
+      if (!Boolean(gsmProject)) {
+        throw new Skipped('Google Secret Manager not configured.')
+      }
+      const client = new SecretManagerServiceClient()
+      const name = `projects/${gsmProject}/secrets/dummy/versions/latest`
+      try {
+        await client.accessSecretVersion({ name })
+      } catch (err) {
+        // no error except gRPC status 5 (i.e. NOT FOUND) is allowed
+        if (!err.code || err.code != 5) {
+          throw Error(`Unable to read a secret. See below for details.\n${err.message}`)
+        }
+      }
+    }, true),
+    check('API token is set', async () => {
+      const token = await appConfig.apiToken
+      if (!Boolean(token)) {
```

Comment:
You should check for the apiToken length. It should be at least 32 characters long

Reply:
fixed